cli/src/runner.rs

```diff
@@ -102,7 +103,25 @@ pub fn build_filesets(
     Ok((file_sets, file_counter))
 }
 
+pub fn get_run_info_id(test: &Test, org_slug: &str, repo: &BundleRepo) -> String {
```

Comment:
Please add test.

Reply:
Also, there are a lot of unnecessary allocations for each ID. Consider re-using buffer which holds input data:
```rust
pub fn get_run_info_id(buf: &mut Vec<u8>, test: &Test, org_slug: &str, repo: &BundleRepo) -> String {
    buf.clear();
    let mut buf = Vec<u8>::new();
    buf.extend_from_slice(org_slug.as_bytes());
    buf.push(b'/');
    buf.extend_from_slice(repo_full_name.as_bytes());
    buf.push(b'/');
    buf.extend_from_slice(test.file.as_deref().unwrap_or("").as_bytes());
    buf.push(b'#');
    buf.extend_from_slice(test.class_name.as_deref().unwrap_or("").as_bytes());
    buf.push(b'#');
    buf.extend_from_slice(test.parent_name.as_bytes());
    buf.push(b'#');
    buf.extend_from_slice(test.name.as_bytes());
    buf.push(b'#');
    buf.extend_from_slice("JUNIT_TESTCASE".as_bytes());
    uuid::Uuid::new_v5(&uuid::Uuid::NAMESPACE_URL, &buf[..]).to_string()
}
```

You can wrap this into `TestIdBuilder` type of you want, which you can create before traversing tests:

```
            let mut id_builder = TestIdBuilder::default();
            for suite in junitxml.suites {
            ...
                      let mut test = Test {
                            parent_name: parent_name.clone(),
                            name: name.clone(),
                            class_name: case.classname.clone(),
                            file: case.file.clone(),
                            id: Some(id_builder.build_test_id(&test, org_slug, repo)),
                        };
```

```